src/ui.cpp

```diff
@@ -895,6 +911,16 @@ void UpdateDialog::StateUpdateAvailable(const Appcast& info, bool installAutomat
         HIDE(m_closeButtonSizer);
         HIDE(m_runInstallerButtonSizer);
         SHOW(m_updateButtonsSizer);
+        if (m_criticalUpdate) {
+            HIDE(m_SkipVersionButtonSizer);
+            HIDE(m_RemindLaterButtonSizer);
+            EnableCloseButton(false);
```

Comment:
> EnableCloseButton(false);

Is it consistent with `(Mac)Sparkle` behavior?

Reply:
I think it isn't - Sparkle [hides skip and later buttons](https://github.com/sparkle-project/Sparkle/blob/180f8975cfc18ef9816f69f963c7463f2f0425b4/Sparkle/SUUpdateAlert.m#L361). That makes sense - you want to prevent the user from ignoring the update and forgetting about it. But being able to close the UI entirely should IMHO be possible, because we don't know the circumstance - the user may e.g. be doing something urgent at the moment.

Reply:
This is a requirement from my workplace. The install is still stoppable, after the download. The user can reject it in UAC prompt, or the later installer itself